src/utils/algolia-queries.js

```diff
@@ -0,0 +1,51 @@
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const escapeStringRegexp = require('escape-string-regexp')
```

Comment:
Q: how does it/does it overlap with existing Algolia stuff?

Reply:
I don't this overlaps with the existing one. 
Because, as I mentioned the product we are using is DocSearch in which we don't do much in terms of setting index as well as rendering the results. 
For indexing, they run crawlers on their server every hour or day to extract data and index accordingly.
For rendering the results, they have provided `docsearch.js` package which is complete and we only need to specify some variables and everything will be good to go.

Reply:
good, thanks ... I understand that those are two different approaches